app/views/devise/mailer/confirmation_instructions.html.erb

```diff
@@ -1,4 +1,4 @@
-<p>Welcome <%= @email %>!</p>
+<p>Welcome <%= @resource.first_name %>!</p>
```

Comment:
We've had some issue in the past that bit us because not every user had a display name: can you change this to `@resource.first_name || @resource.email`?